api/v1alpha1/hosted_controlplane.go

```diff
@@ -90,6 +90,10 @@ type HostedControlPlaneStatus struct {
 	// +kubebuilder:validation:Optional
 	Version string `json:"version,omitempty"`
 
+	// KubeConfig is a reference to the secret containing the default kubeconfig
+	// for this control plane.
+	KubeConfig *corev1.LocalObjectReference `json:"kubeConfig,omitempty"`
```

Comment:
Something this lacks is a way to expose the _key_ in the secret containing the kubeconfig. For example, something like:

```go
type KubeConfigReference struct {
  Secret LocalObjectReference
  Key string
}
```